docs/Developer-documentation-style-guide.md

```diff
@@ -0,0 +1,101 @@
+---
```

Comment:
Is this the header?  I think it's a good habit. 

Would you mind adding it to other documents? thanks